vcd/resource_vcd_vapp_vm.go

```diff
@@ -926,16 +1007,39 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 
 	}
 
-	if d.HasChanges("memory", "cpus", "cpu_cores", "power_on", "disk", "expose_hardware_virtualization",
-		"network", "boot_image", "hardware_version", "os_type", "description") {
+	memoryNeedsColdChange := false
+	cpusNeedsColdChange := false
+	networksNeedsColdChange := false
+	if executionType == "update" {
+		if !d.Get("memory_hot_add_enabled").(bool) && d.HasChange("memory") {
+			memoryNeedsColdChange = true
+		}
+		if !d.Get("cpu_hot_add_enabled").(bool) && d.HasChange("cpus") {
+			cpusNeedsColdChange = true
+		}
+		if isNetworkRemoved(d) && d.HasChange("network") {
+			networksNeedsColdChange = true
+		}
+	} else if len(d.Get("network").([]interface{})) > 0 {
+		// if create happens allow to process it(keeping it as it was), but avoid changing deprecated way of assigning of network
```

Comment:
Please add more commentary of example of what actually is the "deprecated way of assigning of network".

Reply:
added